packages/integration-tests/src/test-project-setup/reference_auth_project.ts

```diff
@@ -0,0 +1,329 @@
+            RefreshToken: 'days',
+            AccessToken: 'minutes',
+            IdToken: 'minutes',
+          },
+          EnableTokenRevocation: true,
+          PreventUserExistenceErrors: 'ENABLED',
+          AllowedOAuthFlows: ['code'],
+          AllowedOAuthScopes: ['openid', 'phone', 'email'],
+          SupportedIdentityProviders: ['COGNITO', 'Facebook', 'Google'],
+          CallbackURLs: ['https://callback.com'],
+          LogoutURLs: ['https://logout.com'],
+          AllowedOAuthFlowsUserPoolClient: true,
+          GenerateSecret: false,
+          ReadAttributes: [
+            'address',
+            'birthdate',
+            'email',
+            'email_verified',
+            'family_name',
+            'gender',
+            'given_name',
+            'locale',
+            'middle_name',
+            'name',
+            'nickname',
+            'phone_number',
+            'phone_number_verified',
+            'picture',
+            'preferred_username',
+            'profile',
+            'updated_at',
+            'website',
+            'zoneinfo',
+          ],
+          WriteAttributes: [
+            'address',
+            'birthdate',
+            'email',
+            'family_name',
+            'gender',
+            'given_name',
+            'locale',
+            'middle_name',
+            'name',
+            'nickname',
+            'phone_number',
+            'picture',
+            'preferred_username',
+            'profile',
+            'updated_at',
+            'website',
+            'zoneinfo',
+          ],
+        });
+
+      const region = await this.cognitoIdentityClient.config.region();
+      const identityPool =
+        await this.authResourceCreator.createIdentityPoolBase({
+          AllowUnauthenticatedIdentities: true,
+          IdentityPoolName: `ref-auth-ip`,
+          AllowClassicFlow: false,
+          CognitoIdentityProviders: [
+            {
+              ClientId: userPoolClient.ClientId,
+              ProviderName: `cognito-idp.${region}.amazonaws.com/${userPool.Id}`,
+              ServerSideTokenCheck: false,
+            },
+          ],
+          SupportedLoginProviders: {
+            'graph.facebook.com': 'clientId',
+            'accounts.google.com': 'clientId',
+          },
+        });
+
+      const roles = await this.authResourceCreator.setupIdentityPoolRoles(
+        userPool.Id!,
+        userPoolClient.ClientId!,
+        identityPool.IdentityPoolId
+      );
+
+      const adminGroup = await this.authResourceCreator.setupUserPoolGroup(
+        'ADMINS',
+        userPool.Id!,
+        identityPool.IdentityPoolId
+      );
+      return {
+        userPool,
+        userPoolClient,
+        domain,
+        identityPool,
+        authRole: roles.authRole,
+        unauthRole: roles.unauthRole,
+        adminGroup,
+      };
+    } catch (e) {
+      await this.authResourceCreator.cleanupResources();
+      throw e;
+    }
```

Comment:
if we make sure that resources are cleanup-script-friendly (which we should anyway) we can remove this catch block

Reply:
TIL, going to look at cleanup script and will update

Reply:
i updated the resource names to be cleanup script friendly